libraries/redcore/factory/factory.php

```diff
@@ -102,4 +102,25 @@ protected static function createDbo()
 
 		return $db;
 	}
+
+	/**
+	 * Return the {@link JDate} object
+	 *
+	 * @param   mixed  $time      The initial time for the JDate object
+	 * @param   mixed  $tzOffset  The timezone offset.
+	 *
+	 * @return  JDate object
+	 *
+	 * @see     JDate
+	 * @since   11.1
+	 */
+	public static function getDate($time = 'now', $tzOffset = null)
+	{
+		if (!$tzOffset)
+		{
+			$tzOffset = new DateTimeZone(self::getConfig()->getValue('offset'));
```

Comment:
also you should consider user timezone... something like
 new DateTimeZone(self::getUser()->getParam('timezone', self::getConfig()->get('offset')));